src/client/content_mapblock.cpp

```diff
@@ -1377,6 +1377,52 @@ void MapblockMeshGenerator::drawNodeboxNode()
 
 	std::vector<aabb3f> boxes;
 	n.getNodeBoxes(nodedef, &boxes, neighbors_set);
+
+	bool isTransparent = false;
+
+	for (const TileSpec &tile : tiles) {
+		if (tile.layers[0].isTransparent()) {
+			isTransparent = true;
+			break;
+		}
+	}
+
+	if (isTransparent) {
+		std::vector<float> sections;
+		sections.reserve(8 + 2 * boxes.size());
+		for (int axis = 0; axis < 3; axis++) {
+			// identify sections
+			sections.clear();
+
+			// Default split at node bounds, up to 3 nodes in each direction
+			for (float s = -3.5f * BS; s < 4.0f * BS; s += 1.0f * BS)
+				sections.push_back(s);
```

Comment:
Might it be more efficient if you preload `sections` with these values outside of the loop and use `sections.resize(n);`?
However I do not know whether this affects the capacity changed by `sections.reserve` above.

Reply:
done